configdialog.cpp

```diff
@@ -578,6 +598,296 @@ void PaletteCustomizeDialog::get_and_save_color(int palette_index) {
+			current_has_errors |= error_name_invalid_characters;
+		}
+		if (current_name.find_first_of(L"\r\n") != std::wstring::npos) {
+			current_has_errors |= error_palette_contains_newlines;
+		}
+
+		import_string_current_index = import_string.find_first_of(L':', import_string_current_index) + 1;
+
+		// Looking for exactly 7 colors, each color starts with # and ends with ; and must contain exactly 6 base digits in base 16
+		for (int i = 0; i < colors_amount; i++) {
+			if (import_string.find_first_of(L':', import_string_current_index) < import_string.find_first_of(L'#', import_string_current_index)) {
+				current_has_errors |= error_colors_too_few;
+				break;
+			}
+
+			if (import_string.find_first_of(L'#', import_string_current_index) == std::wstring::npos) {
+				current_has_errors |= error_colors_too_few;
+				break;
+			}
+			import_string_current_index = import_string.find_first_of(L'#', import_string_current_index) + 1;
+			current_colors[i] = import_string.substr(import_string_current_index, import_string.find_first_of(L';', import_string_current_index) - import_string_current_index);
+			if (import_string.find_first_of(L';', import_string_current_index) == std::wstring::npos) {
+				current_has_errors |= error_colors_too_few;
+				break;
+			}
+			import_string_current_index = import_string.find_first_of(L';', import_string_current_index) + 1;
+
+			if (current_colors[i].length() != 6) {
+				current_has_errors |= error_colors_incorrect_length;
+			}
+			if (current_colors[i].find_first_not_of(valid_hex_chars + L"\r\n") != std::wstring::npos) {
+				current_has_errors |= error_colors_invalid_characters;
+			}
+			if (current_colors[i].find_first_of(L"\r\n") != std::wstring::npos) {
+				current_has_errors |= error_palette_contains_newlines;
+			}
+		}
+
+		parsed_palette_names.push_back(current_name);
+		parsed_palette_colors.push_back(current_colors);
+		parsed_palette_has_errors.push_back(current_has_errors);
+	}
+
+	return new PalettesImport(parsed_palette_names, parsed_palette_colors, parsed_palette_has_errors, has_terminator_at_end);
+}
+
+std::optional<std::wstring> PalettesImport::get_error_string(size_t max_errors) {
+	if (palettes_name.size() == 0) {
+		return std::nullopt;
+	}
+
+	size_t palette_count = palettes_name.size();
+	size_t invalid_palette_count = 0;
+	for (unsigned int has_errors : palettes_has_errors) {
+		if (has_errors != error_none) {
+			invalid_palette_count++;
+		}
+	}
+
+	if (!import_string_had_terminator && invalid_palette_count <= 1 && (invalid_palette_count == 0 || *palettes_has_errors.rbegin() != error_none)) {
+		std::wstring error_string = L"No terminator character (~) found at the end of the input. This may be caused by the inputted text being too long.";
+		auto palette_name = palettes_name.rbegin();
+		if (*palettes_has_errors.rbegin() != error_none) {
+			palette_name++;
+		}
+		error_string.append(L"\nLast valid palette found: " + *palette_name);
```

Comment:
If `palettes_name` only has one element and the previous if statement triggers, then `palette_name` is an invalid iterator that gets dereferenced, causing a crash in debug. You can replicate this by pasting anything that ends with a colon into the palette box.

Reply:
clearly, I haven't learned from the last time I had a crash like this.